### PR TITLE
refractoring of ladders and problems

### LINKED ISSUE #5: [Refactor] Practice and Ladders Page
## Refactoring practice and ladders page.

Here are few practices which we need to ensure while refactoring a file-

1. Removing the inline CSS and adding them into a separate CSS file.
2. Breaking the page into smaller components and reusing them when needed.
3. Adding app the API calls into the actions directory.
4. Making sure the code is modular enough.

Looking forward to your contribution :)

### PR DESCRIPTION
**Pull request by sanyamjain036**
fix #5

### CODE REVIEW

File: src/actions/auth.actions.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export function login(usernameL,passwordL){
+export async function login(usernameL,passwordL){
     return fetch('https://api.codedigger.tech/auth/login/',{
```
Comment (Anurag5086) [inline src/actions/auth.actions.js]: We always use await in an async function.

Comment (Anurag5086) [inline src/actions/auth.actions.js]:
```suggestion
    return await fetch('https://api.codedigger.tech/auth/login/',{
```

File: src/actions/auth.actions.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ export function login(usernameL,passwordL){
       }).then(data => data.json());
 }
 
-export function register (emailR, usernameR, passwordR){
+export async function register (emailR, usernameR, passwordR){
     return fetch(`https://api.codedigger.tech/auth/register/?redirect_url=${process.env.REACT_APP_SEND_EMAIL_RDURL}`,{
```
Comment (Anurag5086) [inline src/actions/auth.actions.js]: Add await here as well and in every async function below.

File: src/actions/lists.actions.js [hunk 3]
```diff
@@ -7,8 +7,17 @@ export function getLists(acc,wise,type){
...
+        method:"GET",
+        headers:{
+            "Content-Type":"application/json",
+            // "Authorization":`Bearer ${acc}`
+        }
+    }).then(data => data.json());
+}
 
-export function getThisList(acc,wise,type,slug){
+export async function getThisList(acc,wise,type,slug){
```
Comment (Anurag5086) [inline src/actions/lists.actions.js]: change `acc` parameter to `accessToken`. Make this change everywhere you find.

File: src/actions/mentors.actions.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ export function getMentors(acc){
     }).then(data => data.json());
 }
 
-export function removeMentor(acc,mname){
+export async function removeMentor(acc,mName){
```
Comment (Anurag5086) [inline src/actions/mentors.actions.js]: change `mName` to `mentorName`

File: src/actions/mentors.actions.js [hunk 5]
```diff
@@ -21,7 +21,7 @@ export function removeMentor(acc,mname){
     }).then(data => data.json());
 }
 
-export function addmentor(acc,guru){
+export async function addmentor(acc,guru){
```
Comment (Anurag5086) [inline src/actions/mentors.actions.js]: Change `guru` to `mentorName` here as well.

File: src/actions/problems.actions.js [hunk 6]
```diff
@@ -7,7 +7,7 @@ export function getProblems(queryStr){
                 }).then(data => data.json());
 } 
 
-export function getProblemsWithCreds(queryStr,acc){
+export async function getProblemsWithCreds(queryStr,acc){
```
Comment (Anurag5086) [inline src/actions/problems.actions.js]: Change `queryStr` to `queryString` and `acc` to `accessToken`.

File: src/actions/profile.actions.js.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-export function getProfile(acc,handle){
+export async function getProfile(acc,handle){
```
Comment (Anurag5086) [inline src/actions/profile.actions.js.js]: change `handle` to `userHandle`.

File: src/components/ProblemLadderCard.js [hunk 8]
```diff
@@ -24,13 +23,13 @@ const LaddersContent = (props) => {
       <>
                 <div className="courses-container">
               <div className="course">
-                <div className={"course-preview " + (bgIndex == 1 ? 'bgIndex1':(bgIndex == 2 ? 'bgIndex2':(bgIndex == 3 ? 'bgIndex3':(bgIndex == 4 ? 'bgIndex4':''))))}>
+                <div className={`course-preview bgIndex${colorNumber}`}>
                   <h6 style={{color:"#d9d9d9"}}>{props.wise1}</h6>
```
Comment (Anurag5086) [inline src/components/ProblemLadderCard.js:25]: Remove all inline css. Add to a new styles.css file.

File: src/pages/ladders/LaddersLevel.js [hunk 9]
```diff
@@ -2,16 +2,16 @@ import React, {useState,useEffect}from 'react'
...
 import ErrorPage from '../ErrorPage/ErrorPage.js';
-
+import {getLists,getListsWithoutAuth} from '../../actions/lists.actions'
 
 function LaddersLevel(props) {
 
     const creds= JSON.parse(localStorage.getItem("creds"));
 
-    const [dat, setDat] = useState();
+    const [dat, setDat] = useState([]);
```
Comment (Anurag5086) [inline src/pages/ladders/LaddersLevel.js]: Rename this to make to more understandable.

**Review by Anurag5086 — CHANGES_REQUESTED** [review]

File: src/pages/ladders/LaddersLevel.js [hunk 10]
```diff
@@ -121,7 +107,8 @@ function LaddersLevel(props) {
                             // console.log(dat);
                             return(
                                 <>
-                                <LaddersContent 
+                                <ProblemLadderCard 
+                                    key={index}
```
Comment (saikeshari) [inline src/pages/ladders/LaddersLevel.js:72]: See if you can combine the two ProblemLadderCards removing the if-else
Try such that pass first_time as parameter with value null when its not available and its value when it is available.
Then check in ProblemLadderCard that if first_time is null, return acc to practice else work acc to ladder.

**Review by saikeshari — CHANGES_REQUESTED** [review]

File: src/actions/mentors.actions.js (hunk 5 above)
Comment (sanyamjain036) [reply src/actions/mentors.actions.js]: but mName is mentorName

Comment (Anurag5086) [reply src/actions/mentors.actions.js]: guru itself is mentor

Comment (sanyamjain036) [reply src/actions/mentors.actions.js]: okk , so, do I have to change mName and guru to mentorName?

Comment (Anurag5086) [reply src/actions/mentors.actions.js]: Yes

**Review by Anurag5086 — APPROVED** [review]
LGTM! 👍

**Comment by Anurag5086** [comment]
@Jeevier There is a conflicting file `src/pages/ladders/LaddersLevel.js`. Try to resolve this conflict.

**Comment by saikeshari** [comment]
Pull the latest code and you will see conflicts @Jeevier .
I have put comments and removed some unused code, Look into and try to resolve.

**Review by saikeshari — APPROVED** [review]

**Pull request merged**